lib/matplotlib/widgets.py

```diff
@@ -1969,24 +1990,56 @@ def __init__(self, ax, onselect, direction, minspan=None, useblit=False,
 
         rectprops['animated'] = self.useblit
 
-        _api.check_in_list(['horizontal', 'vertical'], direction=direction)
-        self.direction = direction
+        self._direction = direction
```

Comment:
Did you mean to use the property setter here? Otherwise, `direction` is not checked on `__init__`.

Reply:
No, I didn't, I suspect that the property setter have been later to allow changing direction of an existing selector and this hasn't been updated.